### PR TITLE
Remove the experimental spaces flag.

### LINKED ISSUE #10061: remove the `spaces_enabled` flag
there's no point having a feature flag which is on by default

### PR DESCRIPTION
**Pull request by clokep**
...to always enable the Spaces features (under unstable endpoints).

Fixes #10061

### CODE REVIEW

File: synapse/federation/transport/server.py [hunk 1]
```diff
@@ -1562,13 +1562,12 @@ def register_servlets(
...
 
-        if hs.config.experimental.spaces_enabled:
-            FederationSpaceSummaryServlet(
-                handler=hs.get_space_summary_handler(),
-                authenticator=authenticator,
-                ratelimiter=ratelimiter,
-                server_name=hs.hostname,
-            ).register(resource)
+        FederationSpaceSummaryServlet(
+            handler=hs.get_space_summary_handler(),
```
Comment (clokep) [inline synapse/federation/transport/server.py:1566]: Note that this gets a different `handler` than the above loop. I have a branch somewhere that changes all of these endpoints to accept a `HomeServer` (like the normal C-S API endpoints) which would probably clean this all up.

**Comment by clokep** [comment]
I've again punted what we want to do with the other experimental flags since those default to off. I'm willing to include removing them here though if we'd like. 🤷

File: synapse/api/room_versions.py [hunk 2]
```diff
@@ -181,6 +181,7 @@ class RoomVersions:
         RoomVersions.V5,
         RoomVersions.V6,
         RoomVersions.MSC2176,
+        RoomVersions.MSC3083,
     )
     # Note that we do not include MSC3083 here unless it is enabled in the config.
```
Comment (richvdh) [inline synapse/api/room_versions.py:186]:
```suggestion
```

**Review by richvdh — APPROVED** [review]
mmkay.

File: synapse/api/room_versions.py (hunk 2 above)
Comment (clokep) [reply synapse/api/room_versions.py:186]: Ugh, thanks.

**Pull request merged**